src/shogun/distance/EuclideanDistance.cpp

```diff
@@ -54,6 +55,32 @@ float64_t CEuclideanDistance::compute(int32_t idx_a, int32_t idx_b)
 		get_feature_vector(idx_b, blen, bfree);
 	ASSERT(alen==blen)
 
+	if(m_dot_enabled)
+	{
+		result+=CMath::dot(avec, bvec, alen);
```

Comment:
found a performance issue with linalg::dot and sgvector for smaller dimensions. https://gist.github.com/lambday/7cbd77eb0530ad41d9f5 . So we might have to use cmath::dot for time being till we have updates as this is inside `compute()`